dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/UserDeletionHandler.java

```diff
@@ -72,11 +74,21 @@ private void deleteUserRole( UserRole role )
 
     private void deleteOrganisationUnit( OrganisationUnit unit )
     {
-        for ( User user : unit.getUsers() )
+        for ( User user : userStore.getUsers( new UserQueryParams().addOrganisationUnit( unit ) ) )
```

Comment:
FYI: test showed that `unit.getUsers()` would not work. I assume this is because the object is not fully initialized.

Reply:
Right, it depends where the incoming org unit is coming from. If it comes from outside the "transactional boundary" / not within a transaction/Hibernate session then the users will not be initialized. I think for most deletion handlers, the incoming object will be attached to a transaction/session. So I guess it would be good to understand why this is a problem here in particular instead of working around. Could be a good reason for all I know.

Reply:
The instance came directly from test setup where it never was fully initialized by hibernate. Also there was another issue in the setup causing a `null` to be inserted in the set of units which I fixed now.

Reply:
Great!